### DISCUSSION TITLE
How to Develope/Contribute ?

### DISCUSSION THREAD

**Discussion by ImanEmadi** [apache/camel-karavan#1227] [Q&A] [answered]
### Ask a question

Hello,
I need to add new and modify existing features, but I am having difficulties how to get started.

there are thousands lines of code, with no comments, documentation and API reference about what each part is doing. 
any guide/tips regarding how to figure things out in the project?
for example, I can't find out how to access the selected file by the user, in the designer directory, like for writing something new into the file after they have interacted with the GUI.

### Variant

Vs Code Extension

### Container Management (if applicable)

None

### Operating System (if applicable)

Linux

### Version

4.5.0

**Comment by davsclaus** [comment, marked as answer]
As any open source project of resonable size it requires devotion and skills to work on. 
I am sorry but there is no other way than hard work.

**Reply by ImanEmadi** [reply]
It indeed requires devotion.
not having proper documentation for thousands lines of code, is definitely not a good practice and reduces the number of people that can potentially contribute to the project.
in this case, i would rather call it 'time consuming' than 'hard work'.

**Reply by anandroy9279-pixel** [reply]
I know so it is good product
